app/controllers/datasets_controller.rb

```diff
@@ -1,8 +1,9 @@
 class DatasetsController < ApplicationController
   before_action :authenticate_user!
+  before_action :set_dataset, only: [:show, :edit, :update, :destroy,
```

Comment:
That's a lot of onlies :) you can use `except` instead to exclude things

Reply:
Yeah not great indeed, but I'd rather maintain a whitelist than a blacklist; I don't want the callback to automatically apply to any new controller actions.

I was also considering removing the callback altogether and having a `set_dataset` call in all the controller actions that need it.

Reply:
That makes sense 👍